### ISSUE TITLE
Add TLS support

### ISSUE THREAD

**Issue by ryan-summers** [quartiq/minimq#121]
Add support for a TLS-based connection with the broker.

This may necessitate the development of new `embedded-nal` traits for the TCP stack.

[`embedded-tls`](https://github.com/drogue-iot/embedded-tls) can provide the necessary TLS components.

**Comment by teovillanueva** [comment]
Hello! Is TLS support planned?

**Comment by ryan-summers** [comment]
It is indeed planned eventually, but much of minimq development is driven by the user needs of https://github.com/quartiq/stabilizer and https://github.com/quartiq/booster. Given that we don't have any immediate need yet, it's not yet on my priority list and is more of a fun project I'd like to get to later.

**Comment by Ddystopia** [comment]
Hello, I want to mention https://github.com/drogue-iot/embedded-tls here. It allows to wrap any tcp stream into tls. Maybe it would be useful for someone

**Comment by ryan-summers** [comment]
Yeah, the intent would be to leverage `embedded-tls` (or `rustls` when no-std support is complete) to facilitate this

**Comment by overheat** [comment]
`embedded-tls` already support Client Certificate Authorization, https://github.com/drogue-iot/embedded-tls/pull/135
Cloud we have this tls support with a mutual auth example, like AWS iot core does?

**Comment by jordens** [comment]
#173 etc now provide async and a embedded-tls example.
